src/connector/src/sink/elasticsearch_opensearch/elasticsearch_opensearch_config.rs

```
@@ -110,27 +115,36 @@ impl ElasticSearchOpenSearchConfig {
         let url =
             Url::parse(&self.url).map_err(|e| SinkError::ElasticSearchOpenSearch(anyhow!(e)))?;
         if connector.eq(ES_SINK) {
-            let transport = elasticsearch::http::transport::TransportBuilder::new(
+            let mut transport_builder = elasticsearch::http::transport::TransportBuilder::new(
                 elasticsearch::http::transport::SingleNodeConnectionPool::new(url),
-            )
-            .auth(elasticsearch::auth::Credentials::Basic(
-                self.username.clone(),
-                self.password.clone(),
-            ))
-            .build()
-            .map_err(|e| SinkError::ElasticSearchOpenSearch(anyhow!(e)))?;
+            );
+            if let Some(username) = &self.username
+                && let Some(password) = &self.password
+            {
+                transport_builder = transport_builder.auth(
+                    elasticsearch::auth::Credentials::Basic(username.clone(), password.clone()),
+                );
+            }
```

Should we throw an error if only password or only username is provided?